### PR TITLE
Perform validation after migration on serialtiles and mbtiles

### LINKED ISSUE #71: Allow upgrading crufty v1 tiles in migration stream
We gained the ability to migrate vector tiles adhering to the v1 spec version to v2 in #62.

But a problem exists:

  - The v1 spec was not strict (or at least its implementations) about layers with no features.
  - Validation is done [currently in migration-stream.js **before** upgrading](https://github.com/mapbox/mapbox-tile-copy/blob/cca8e80fc57811e7b53e199c8a03a6f7625d2379/lib/migration-stream.js#L37-L65)
  - The validation code currently throws on layers with no features (which are harmless and common). 
  - This means that v1 tiles with empty layers cannot be upgraded and the whole process bails.

Instead, for usecases of `mapbox-tile-copy` like batch migrating large tilesets to v2, we want:

  - Any v1 tile that can be upgraded, is upgraded
  - The resulting v2 tile should be valid

/cc @mapsam @ajashton @GretaCB

### PR DESCRIPTION
**Pull request by jakepruitt**
Performs validation after migration from v1 to v2. Fixes #71 

cc/ @springmeyer @mapsam @GretaCB @ajashton

### CODE REVIEW

File: lib/migration-stream.js [hunk 1]
```diff
@@ -36,37 +36,60 @@ function MigrationStream() {
...
+              break;
+            }
+          }
+        } else {
+          err = new Error('Invalid data');
+        }
+        err.code = 'EINVALID';
+        return callback(err);
+      }
+
```
Comment (mapsam) [inline lib/migration-stream.js]: @jakepruitt wondering if we could turn this `if(info.errors)` check into a function now that we are using it twice?

Comment (mapsam) [inline lib/migration-stream.js]: something like

```
ifErrors(info, function(err) {
  if (err) return callback(err);
  migrationStream.push(tile);
  return callback();
});
```

File: test/copy.tilelive.test.js [hunk 2]
```diff
@@ -278,16 +278,17 @@ test('copy corrupt mbtiles', function(t) {
   });
 });
 
-test('error on invalid tile in mbtiles', function(t) {
+test('passes through invalid tile in mbtiles', function(t) {
```
Comment (mapsam) [inline test/copy.tilelive.test.js]: Just for clarity's sake, why is this mbtile file invalid? Would be nice to reference this in the future if we are more explicit with the test name / fixture name.

File: lib/migration-stream.js (hunk 1 above)
Comment (jakepruitt) [inline lib/migration-stream.js]: Good call, we can probably do this on the tile itself, and do it synchronously.

File: test/copy.tilelive.test.js [hunk 3]
```diff
@@ -279,11 +279,11 @@ test('copy corrupt mbtiles', function(t) {
 });
 
 test('passes through invalid tile in mbtiles', function(t) {
-  var fixture = path.resolve(__dirname, 'fixtures', 'invalid.tiles.mbtiles');
+  var fixture = path.resolve(__dirname, 'fixtures', 'invalid.tile-with-no-geometry.mbtiles');
```
Comment (mapsam) [inline test/copy.tilelive.test.js]: 🙏 thanks @jakepruitt!

**Comment by springmeyer** [comment]
👍 Looks great @jakepruitt - @ajashton: does this get you past all the errors you were seeing?

**Comment by jakepruitt** [comment]
Works! @ajashton your tilejson now copies and gets migrated 🙇 . I think we'll PR the tilejson migration separately and possibly add it as a configurable option in the executable.

**Pull request merged**